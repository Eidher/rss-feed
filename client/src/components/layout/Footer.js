import React, { Component } from "react";
import { Link } from "react-router-dom";

class Footer extends Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <footer className="w-100">
        <div className="container">
          <p>
            &copy; RSS Feed - Eidher Escalona - {new Date().getFullYear()}. All
            Rights Reserved.
          </p>
          <ul className="list-inline">
            <li className="list-inline-item">
              <Link to="/">Home</Link>
            </li>
            <li className="list-inline-item">
              <Link to="/feed">RSS Feed</Link>
            </li>
            <li className="list-inline-item">
              <Link to="/contact">Contact</Link>
            </li>
          </ul>
        </div>
      </footer>
    );
  }
}

export default Footer;
