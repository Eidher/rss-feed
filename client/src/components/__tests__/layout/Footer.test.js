import React from "react";
import { shallow } from "enzyme";
import Footer from "../../layout/Footer";

describe("<Footer />", () => {
  const wrapper = shallow(<Footer />);

  it("renders correctly.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should not update component", () => {
    const component = new Footer({});
    expect(component.shouldComponentUpdate()).toBe(false);
  });

  it("should have the right Links and locations", () => {
    expect(wrapper).toContainMatchingElements(3, "Link");
    wrapper.find("Link").forEach(node => {
      expect(node).toHaveProp("to");
    });
  });
});
