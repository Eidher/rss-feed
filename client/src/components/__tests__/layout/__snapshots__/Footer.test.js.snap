// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly. 1`] = `
<footer
  className="w-100"
>
  <div
    className="container"
  >
    <p>
      Â© RSS Feed - Eidher Escalona - 
      2019
      . All Rights Reserved.
    </p>
    <ul
      className="list-inline"
    >
      <li
        className="list-inline-item"
      >
        <Link
          to="/"
        >
          Home
        </Link>
      </li>
      <li
        className="list-inline-item"
      >
        <Link
          to="/feed"
        >
          RSS Feed
        </Link>
      </li>
      <li
        className="list-inline-item"
      >
        <Link
          to="/contact"
        >
          Contact
        </Link>
      </li>
    </ul>
  </div>
</footer>
`;
